### PR TITLE
Add HCP Consul Plus Tier

### PR DESCRIPTION
**Pull request by i0rek**
### :hammer_and_wrench: Description

Pulling in https://github.com/hashicorp/hcp-sdk-go/pull/24 to enable using private beta `plus` tier. Use `go generate` to regenerate docs.

### :building_construction: Acceptance tests

- [x] Are there any feature flags that are required to use this functionality? https://github.com/hashicorp/cloud-consul-service/pull/1164 contains the two flags that were added to control `plus` tier.
- [ ] Have you added an acceptance test for the functionality being added?
- [x] Have you run the acceptance tests on this branch?

Output from acceptance testing:

I had to change the tier and scale in the test before this was passing.

```
$ make testacc TESTARGS='-run=TestAccConsulCluster'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./internal/... -v -run=TestAccConsulCluster -timeout 120m
?   	github.com/hashicorp/terraform-provider-hcp/internal/clients	[no test files]
testing: warning: no tests to run
PASS
ok  	github.com/hashicorp/terraform-provider-hcp/internal/consul	(cached) [no tests to run]
testing: warning: no tests to run
PASS
ok  	github.com/hashicorp/terraform-provider-hcp/internal/input	(cached) [no tests to run]
=== RUN   TestAccConsulCluster
--- PASS: TestAccConsulCluster (1178.55s)
PASS
ok  	github.com/hashicorp/terraform-provider-hcp/internal/provider	1179.231s
...
```

### CODE REVIEW

File: internal/provider/validators_test.go [hunk 1]
```diff
@@ -267,6 +267,18 @@ func Test_validateConsulClusterTier(t *testing.T) {
 		input    string
 		expected diag.Diagnostics
 	}{
+		"development": {
+			input:    "development",
+			expected: nil,
+		},
```
Comment (riddhi89) [inline internal/provider/validators_test.go:273]: looks like the `valid tier lowercase` can be removed with this test case doing the same thing.

File: GNUmakefile [hunk 2]
```diff
@@ -55,6 +55,7 @@ depscheck:
 gencheck:
 	@echo "==> Checking generated source code..."
 	go generate
+	@git diff
```
Comment (bcmdarroch) [inline GNUmakefile:58]: is this just to show the diff?

File: templates/resources/consul_cluster.md.tmpl [hunk 3]
```diff
@@ -14,6 +14,7 @@ description: |-
 {{ tffile "examples/resources/hcp_consul_cluster/resource.tf" }}
 
 -> **Note:** The `primary_link` attribute is related to federation, a feature that is currently in private beta.
+-> **Note:** The `plus` tier is currently in private beta.
```
Comment (bcmdarroch) [inline templates/resources/consul_cluster.md.tmpl:17]: Just double-checked this with [the doc preview tool](https://registry.terraform.io/tools/doc-preview) and looks like there needs to be a blank line between the notes for them to render separately.

Also, could you move the banners below `# {{.Type}} ({{.Name}})`? I just realized we don't have a consistent pattern across our pages. 😔

File: internal/provider/resource_consul_cluster.go [hunk 4]
```diff
@@ -69,7 +69,7 @@ func resourceConsulCluster() *schema.Resource {
 			},
 			"tier": {
 				// TODO: link to HCP Consul feature tier page when it is available
-				Description:      "The tier that the HCP Consul cluster will be provisioned as.  Only `development` and `standard` are available at this time.",
+				Description:      "The tier that the HCP Consul cluster will be provisioned as.  Only `development`, `standard` and `plus` are available at this time.",
```
Comment (bcmdarroch) [inline internal/provider/resource_consul_cluster.go:72]: Looks like the pricing page exists now if you want to add the link here. 
```suggestion
				Description:      "The tier that the HCP Consul cluster will be provisioned as.  Only `development`, `standard` and `plus` are available at this time. See [pricing information](https://cloud.hashicorp.com/pricing/consul).",
```

**Review by bcmdarroch — APPROVED** [review]
Looks good! 🎉 Just had some documentation nits and the new go SDK version that can go in go.mod.

File: internal/provider/validators_test.go (hunk 1 above)
Comment (i0rek) [reply internal/provider/validators_test.go:273]: I like having the additional test case. I know it is duplicated, but makes sure all tiers are covered and the different ways to spell them.

**Review by riddhi89 — APPROVED** [review]

File: docs/resources/consul_cluster.md [hunk 5]
```diff
@@ -26,15 +30,14 @@ resource "hcp_consul_cluster" "example" {
...
--> **Note:** The `primary_link` attribute is related to federation, a feature that is currently in private beta.
 <!-- schema generated by tfplugindocs -->
 ## Schema
 
 ### Required
 
 - **cluster_id** (String) The ID of the HCP Consul cluster.
 - **hvn_id** (String) The ID of the HVN this HCP Consul cluster is associated to.
-- **tier** (String) The tier that the HCP Consul cluster will be provisioned as.  Only `development` and `standard` are available at this time.
+- **tier** (String) The tier that the HCP Consul cluster will be provisioned as.  Only `development`, `standard` and `plus` are available at this time.
```
Comment (riddhi89) [inline docs/resources/consul_cluster.md:40]: I think you might need to regen the docs to include the added link to the pricing page.

**Comment by i0rek** [comment]
@bcmdarroch I addressed your feedback. I was just wondering if I need to add any tests?

**Comment by bcmdarroch** [comment]
I think the validator test update is sufficient as far as unit tests go. I would do a local run of the Consul cluster acceptance test swapping `development` for `plus` and add the results to your PR description - but probably best to leave the acceptance tests on `development` since that doesn't require billing set up on the test environment.

**Comment by i0rek** [comment]
Put the output in the PR description.

**Pull request merged**